src/main/java/net/sf/jabref/logic/exporter/BibDatabaseWriter.java

```diff
@@ -158,6 +158,8 @@ public E savePartOfDatabase(BibDatabaseContext bibDatabaseContext,
 
         session = saveSessionFactory.createSaveSession(preferences.getEncodingOrDefault(), preferences.getMakeBackup());
 
+        writeDatabaseID(bibDatabaseContext.getDatabase().getDatabaseID());
```

Comment:
Only write it if we don't want to save in plan bibtex mode (see below, line 198)